src/components/appheader.main.jsx

```diff
@@ -149,17 +152,18 @@ class AppHeaderMain extends React.Component {
             <AppHeaderLoginMain isMobileView={false} />
           </div>
 
-          <div className="cart-link-container">
-            <Link className="cart-link" to="/mybag">
-              {cartData && cartData['total-quantity'] !== 0 && !isLoading && (
-                <span className="cart-link-counter">
-                  {cartData['total-quantity']}
-                </span>
-              )}
-              {intl.get('shopping-bag-nav')}
-            </Link>
-          </div>
-
+          {!isB2B && !Config.b2b.enable && (
```

Comment:
This condition won't work as expected. We need to show these components only when we're in B2C, or only in B2B after a buyer has logged in.

Something like the following should work better:
`{!Config.b2b.enable || (Config.b2b.enable && isB2B) && (....`